base/utils/R/combine_model_netcdf.R

```diff
@@ -0,0 +1,167 @@
+#' Merge model outputted netCDF files by time steps specified by pecan settings file.
+#' @details
+#' The function is only tested for SIPNET model runs.
+#' Please make sure you have the same netCDF formats if you want to proceed with different models.
+#' We could also have more functions that deal with different dimensions (e.g., by site instead of by year).
+#' 
+#' @param settings.dir character: physical path to the pecan standard multi-settings file.
+#' @param nc.outdir  character: physical path to the folder that contains the merged netCDF files.
+#'
+#' @return character: file paths to the merged netCDF files.
+#' @export
+#' 
+#' @author Dongchen Zhang
+#' @importFrom purrr %>%
+#' @importFrom foreach %dopar%
+all_site_nc_merge_by_year <- function (settings.dir, nc.outdir) {
+  # check shell environments.
+  if ("try-error" %in% class(try(temp <- system("which cdo", intern = T), silent = T))) {
```

Comment:
This doesn't work on my machine --  when cdo isn't found, `which` returns a warning not an error, and therefore isn't caught by `try`. Maybe...?

```suggestion
  if (suppressWarnings(system2("which", "cdo", stdout = FALSE)) != 0 ) {
```

Reply:
Fixed. In addition, I also fixed the checks for the `merge_image_tiles` functions.